app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+     * @param priority The priority of the message. The param is a int and the lower the number the
+     *                 higher priority the message has. 0 is the highest.
+     * @return the id of the message.
+     */
+    public int message(String message, int priority) {
+        return message(message, null, null, priority, true, 0);
+    }
+
+    /**
+     * Wrapper to the full message method {@link #message(String, String, View.OnClickListener, int,
+     * boolean, int)}.
+     * The message will be dismissible and won't have timer.
+     * @param message The message String.
+     * @param actionMessage The label for the action button.
+     * @param actionOnClick The View.OnClickListener for the action button.
+     * @param priority The priority of the message. The param is a int and the lower the number the
+     *                 higher priority the message has. 0 is the highest.
+     * @return the id of the message
+     */
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                       int priority){
+        return message(message, actionMessage, actionOnClick, priority, true, 0);
+    }
+
+    /**
+     * Add to the list and display a new message. This is the method that should be used to
+     * display messages and it's being called by {@link BaseActivity#snackBar}.
+     * @param message The message String.
+     * @param actionMessage The label for the action button.
+     * @param actionOnClick The View.OnClickListener for the action button.
+     * @param priority The priority of the message. The param is a int and the lower the number the
+     *                 higher priority the message has. 0 is the highest.
+     * @param isDismissible if true the message will have a X button to remove it self from the
+     *                      list.
+     * @param secondsToTimeOut Number of seconds to message auto dismiss. 0 to never.
+     * @return the id of the message.
+     */
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                        int priority, boolean isDismissible, int secondsToTimeOut){
+        mMessageId++;
+        MessageKey key = new MessageKey(mMessageId, priority);
+        Message value = new Message(key, message, actionMessage, actionOnClick, isDismissible);
+        mMessagesList.put(key, value);
+        updateList();
+        setTimer(key, secondsToTimeOut);
+        if (mMessagesList.size() == 0) {
+            hide();
+        } else {
+            show();
+        }
+        return mMessageId;
+    }
+
+    /**
+     * Sets the auto-dismiss timer to the message given it's key and seconds to dismiss.
+     * @param key The message key.
+     * @param seconds Seconds until dismiss.
+     */
+    private void setTimer(final MessageKey key, int seconds) {
+        if(seconds > 0) {
+            int limit = seconds*1000;
+            new CountDownTimer(limit, 1000) {
```

Comment:
With this usage of `CountDownTimer`, you'll get onTick calls every second even though you're not using them. I suggest using `new CountDownTimer(limit, limit)` instead. That way, it will only tick once at the end, and be more efficient.